src/ActionModal/components/ActionModal.vue

```diff
@@ -389,6 +389,12 @@ export default {
       } catch ({ message }) {
         this.submissionError = `${this.submissionErrorPrefix}: ${message}.`
       }
+
+      // limit fees to the maximum the user has
+      if (this.invoiceTotal > this.balanceInAtoms) {
+        this.gasPrice =
+          (this.balanceInAtoms - Number(this.amount)) / this.gasEstimate
```

Comment:
if `balanceInAtoms` is 0, won't this result in a negative number?

Reply:
good question! We should add a comment. `this.amount` can't be bigger then `balanceInAtoms` at this point as there is a check on the first page.